pennylane/transforms/resolve_dynamic_wires.py

```diff
@@ -215,17 +216,23 @@ def multiple_allocations():
                 deallocated.add(w)
                 manager.return_wire(wire_map.pop(w))
         else:
-            op = op.map_wires(wire_map)
+            if wire_map:
+                op = op.map_wires(wire_map)
             if intersection := deallocated.intersection(set(op.wires)):
-                raise ValueError(
+                raise AllocationError(
                     f"Encountered deallocated wires {intersection} in {op}. Dynamic wires cannot be used after deallocation."
                 )
-            new_ops.append(op.map_wires(wire_map))
```

Comment:
Wait, was the mapping in there twice? 😅 
If so, do we need to improve tests for mapping?